### PR TITLE
Add fade option to carousel and fix glitches

### PR DESCRIPTION
**Pull request by mjlescano**
Closes https://github.com/Synthetixio/gmv3-interface/issues/9

### CODE REVIEW

File: src/components/Carousel.tsx [hunk 1]
```diff
@@ -35,112 +38,97 @@ export default function Carousel({
 			})
 		)
 	);
-	const startingIndex = () => {
-		if ((carouselItems.length & 2) === 0) {
-			return Math.ceil(carouselItems.length / 2);
-		} else {
-			return Math.ceil(carouselItems.length / 2);
-		}
```
Comment (mjlescano) [inline src/components/Carousel.tsx:43]: Removed the `if` here as it was setting the same value, and updated it to only select the middle item if there are more than `2` items.

File: src/stories/Carousel.stories.tsx [hunk 2]
```diff
@@ -1,5 +1,7 @@
 import React from 'react';
```
Comment (fritzschoff) [inline src/stories/Carousel.stories.tsx:1]: Can you update the name of the file accordingly to the component. FadedCarousel

File: src/components/Carousel.tsx [hunk 3]
```diff
@@ -22,9 +24,10 @@ export default function Carousel({
 	maxWidth,
 	withArrows = true,
 	withDots = true,
+	withFade = false,
 	carouselItems,
 	...rest
-}: CarouselProps) {
+}: FadedCarouselProps) {
```
Comment (fritzschoff) [inline src/components/Carousel.tsx:30]: Can you rename the component as well please

**Review by fritzschoff — APPROVED** [review]

File: src/stories/Carousel.stories.tsx (hunk 2 above)
Comment (mjlescano) [reply src/stories/Carousel.stories.tsx:1]: At the beginning I renamed it to `FadedCarousel`, but ended up adding the `withFade` option to the original `Carousel`. The code added looked fairly straightforward, what do you think?

File: src/components/Carousel.tsx (hunk 3 above)
Comment (mjlescano) [reply src/components/Carousel.tsx]: ReFixed to `CarouselProps` as the renaming didn't take place.

**Pull request merged**